src/alire/alire-origins-deployers-git.adb

```diff
@@ -1,15 +1,68 @@
+with Alire.Directories;
+with Alire.Errors;
+with Alire.OS_Lib.Subprocess;
+
 with Alire.VCSs.Git;
 
 package body Alire.Origins.Deployers.Git is
 
-   -----------
-   -- Fetch --
-   -----------
+   ------------
+   -- Deploy --
+   ------------
 
    overriding
    function Deploy (This : Deployer; Folder : String) return Outcome is
    begin
       return VCSs.Git.Handler.Clone (This.Base.URL_With_Commit, Folder);
    end Deploy;
 
+   ------------------
+   -- Compute_Hash --
+   ------------------
+
+   overriding
+   function Compute_Hash (This   : Deployer;
+                          Folder : String;
+                          Kind   : Hashes.Kinds) return Hashes.Any_Digest
+   is
+      pragma Unreferenced (This);
+      use OS_Lib.Subprocess;
+
+      --  Enter the folder to hash
+      Guard : Directories.Guard (Directories.Enter (Folder)) with unreferenced;
+
+      --  Determine the coreutils tool to use for hashing
+      Hasher : constant String := Utils.To_Lower_Case (Kind'Img) & "sum";
+      --  E.g.; sha512sum
+
+      --  Let git do its magic
+      Output    : Utils.String_Vector;
+      Exit_Code : Integer;
+   begin
+      Raw_Spawn (Program    => "sh",
```

Comment:
Why use `sh` here? Is it possible to spawn `git` directly?

Reply:
I don't know a way to pipe stdout of a subprocess to stdin with GNAT.Os_Lib, but I might have missed it. Do you know if GNAT has portable support for some kind of pipes?

Alternatively, we can use a temporary file.

Reply:
Done with temporary file (commit 7f10d7c).